### PR TITLE
machine-api-provider-openstack: Add proxy test

### PR DESCRIPTION
**Pull request by pierreprinetti**
Add an optional, on-demand pre-submit test exercising proxy
installation.

Tests [Bug 2046133](https://bugzilla.redhat.com/show_bug.cgi?id=2046133)

### CODE REVIEW

File: ci-operator/config/openshift/machine-api-provider-openstack/openshift-machine-api-provider-openstack-master.yaml [hunk 1]
```diff
@@ -71,6 +71,16 @@ tests:
       CUSTOM_FEATURE_FLAGS_ENABLED: MachineAPIProviderOpenStack
       TEST_SUITE: experimental/reliability/minimal
     workflow: openshift-e2e-openstack-ipi
+- as: e2e-openstack-proxy
+  optional: true
+  steps:
+    cluster_profile: openstack-vh-mecha-central
+    env:
+      BASE_DOMAIN: shiftstack.devcluster.openshift.com
```
Comment (EmilienM) [inline ci-operator/config/openshift/machine-api-provider-openstack/openshift-machine-api-provider-openstack-master.yaml:79]: you don't need that, it's defined in the workflow already.

File: ci-operator/config/openshift/machine-api-provider-openstack/openshift-machine-api-provider-openstack-master.yaml [hunk 2]
```diff
@@ -71,6 +71,16 @@ tests:
...
+      CONFIG_TYPE: minimal
```
Comment (EmilienM) [inline ci-operator/config/openshift/machine-api-provider-openstack/openshift-machine-api-provider-openstack-master.yaml:80]: no, it's "proxy", and is defined in the workflow already.

File: ci-operator/config/openshift/machine-api-provider-openstack/openshift-machine-api-provider-openstack-master.yaml [hunk 3]
```diff
@@ -71,6 +71,16 @@ tests:
...
+      CUSTOM_FEATURE_FLAGS_ENABLED: MachineAPIProviderOpenStack
+      TEST_SUITE: experimental/reliability/minimal
+    workflow: openshift-e2e-openstack-proxy
```
Comment (EmilienM) [inline ci-operator/config/openshift/machine-api-provider-openstack/openshift-machine-api-provider-openstack-master.yaml:83]: you don't need that, it's defined in the workflow already.

Comment (pierreprinetti) [reply ci-operator/config/openshift/machine-api-provider-openstack/openshift-machine-api-provider-openstack-master.yaml:83]: I assume you refer to the test suite. OK. Incidentally, I don't think `minimal` should be defined in the workflow, but nopt a topic for this PR.

**Pull request merged**